test/unit/module/core/simd_cast.cpp

```diff
@@ -102,12 +103,23 @@ template<typename T, typename U>
 void
 simd_cast_test()
 {
-  // saving compile time a bit
-  if constexpr( T::size() * 2 <= U::size() )
+  if constexpr( eve::current_api == eve::rvv )
   {
-    simd_cast_test_one_way<T, U>();
-    simd_cast_test_one_way<U, T>();
+    // for RISC-V we used decrease number of checks greatly, so
+    // default condition will not work properly
+    if constexpr( sizeof(T) * 2 <= sizeof(U) )
```

Comment:
This check for compile time reduction doesn't strike me as good.

You want at least up to sizeof(T) * 2 <= sizeof(U) * 2. 

ALso - what's with "default condition will not work properly"

Finally - both if else are the same.

Reply:
> ALso - what's with "default condition will not work properly"

By default  `if constexpr( T::size() * 2 <= U::size() )` is used, but it seems `wide::size()` return cardinal number:

https://github.com/jfalcou/eve/blob/ed5347d230d34a7f59a9d6ca32257603c10b57ab/include/eve/arch/cpu/wide.hpp#L377

Here I'm checking conversions between same sizes wide, and the cases when we need get size twice bigger or twice smaller, but cardinal sizes for `float` and `std::int8_t` will be too different